src/indexable.cr

```diff
@@ -166,7 +166,7 @@ module Indexable(T)
   # [2, 5, 7, 10].bsearch_index { |x, i| x >= 4 } # => 1
   # [2, 5, 7, 10].bsearch_index { |x, i| x > 10 } # => nil
   # ```
-  def bsearch_index
+  def bsearch_index(&block : T, Int32 -> Bool)
```

Comment:
Since this is returning an index should the return type also be restricted?

```suggestion
  def bsearch_index(&block : T, Int32 -> Bool) : Int32
```

Reply:
I guess so, but it has nothing to do with the intention of this PR.

Reply:
Just thought I would mention it :)